lib/modules/manager/mix/artifacts.spec.ts

```diff
@@ -291,6 +303,45 @@ describe('modules/manager/mix/artifacts', () => {
     ]);
   });
 
+  it('does not add registries configured in registryAliases but no token in hostRules', async () => {
+    GlobalConfig.set({ ...adminConfig, binarySource: 'install' });
+    fs.readLocalFile.mockResolvedValueOnce('Old mix.lock');
+    fs.findLocalSiblingOrParent.mockResolvedValueOnce('mix.lock');
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce('New mix.lock');
+    hostRules.getAll.mockReturnValueOnce([
+      { matchHost: 'https://getoban.pro/repo' },
+    ]);
+    hostRules.find.mockReturnValueOnce({});
+
+    // erlang
+    getPkgReleases.mockResolvedValueOnce({
+      releases: [{ version: '25.0.0.0' }],
+    });
+    // elixir
+    getPkgReleases.mockResolvedValueOnce({
+      releases: [{ version: 'v1.13.4' }],
+    });
+
+    const result = await updateArtifacts({
+      packageFileName: 'mix.exs',
+      updatedDeps: [{ depName: 'oban_pro' }],
+      newPackageFileContent: '{}',
+      config,
+    });
+
+    expect(result).toEqual([
+      {
+        file: { type: 'addition', path: 'mix.lock', contents: 'New mix.lock' },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
```

Comment:
i don't see a test for the `mix repo.add ...` command

Reply:
We aren't taking credentials and logging them in. The user provides the auth token generated by doing that command locally and which is what is passed in the auth header. 

`mix repo.add` takes a set of credentials and other arguments and stores an auth token on the local machine. When a dependency with a repo that isn't `hexpm` is encountered, it looks up the token and URL from that config. Rather than inventing and introducing a new auth system to renovate, we use the package rule to look up the URL and the host rule then adds the token to the request.

The readme lays out the configuration for the user pretty clearly. Let me know if something is missing there

Reply:
then add a test for it